### PR TITLE
[PYTHON][CAPI][AUTO] Add ENABLE_STARTUP_FALLBACK and ENABLE_RUNTIME_FALLBACK proper…

### PR DESCRIPTION
**Pull request by WeldonWangwang**
### Details:
  - Add ENABLE_STARTUP_FALLBACK and ENABLE_RUNTIME_FALLBACK and DEVICE_BIND_BUFFER properties test cases

### Tickets:
 - CVS-106021

### CODE REVIEW

File: src/bindings/python/src/pyopenvino/core/properties/properties.cpp [hunk 1]
```diff
@@ -95,6 +95,14 @@ void regmodule_properties(py::module m) {
 
     wrap_property_RW(m_intel_gpu, ov::intel_gpu::enable_loop_unrolling, "enable_loop_unrolling");
 
+    // Submodule auto
+    py::module m_intel_auto =
+        m_properties.def_submodule("intel_auto",
+                                   "openvino.runtime.properties.intel_auto submodule that simulates ov::intel_auto");
+
+    wrap_property_RW(m_intel_auto, ov::intel_auto::enable_startup_fallback, "enable_startup_fallback");
+    wrap_property_RW(m_intel_auto, ov::intel_auto::enable_runtime_fallback, "enable_runtime_fallback");
```
Comment (songbell) [inline src/bindings/python/src/pyopenvino/core/properties/properties.cpp:104]: shall we add bind_buffer also?

File: src/bindings/python/tests/test_runtime/test_properties.py [hunk 2]
```diff
@@ -247,6 +247,16 @@ def test_properties_ro(ov_property_ro, expected_value):
                 (2.0, 2.0),
             ),
         ),
+        (
+            properties.intel_auto.enable_startup_fallback,
+            "ENABLE_STARTUP_FALLBACK",
+            ((True, True),),
```
Comment (wangleis) [inline src/bindings/python/tests/test_runtime/test_properties.py:253]: it is better to add all cases like 
            (
                (True, True),
                (False, False),
                (1, True),
                (0, False),
            ),

File: src/bindings/python/src/pyopenvino/core/properties/properties.cpp (hunk 1 above)
Comment (WeldonWangwang) [reply src/bindings/python/src/pyopenvino/core/properties/properties.cpp:104]: Yes, we can add it.

File: src/bindings/python/tests/test_runtime/test_properties.py (hunk 2 above)
Comment (WeldonWangwang) [reply src/bindings/python/tests/test_runtime/test_properties.py:253]: Okay.

File: src/bindings/c/include/openvino/c/ov_property.h [hunk 3]
```diff
@@ -171,3 +171,25 @@ ov_property_key_enable_profiling;
...
+/**
+ * @brief Read-write property<string> for setting that enables performance improvement by binding
+ * buffer to hw infer request
+ * @ingroup ov_property_c_api
+ */
+OPENVINO_C_VAR(const char*)
+ov_property_key_intel_auto_device_bind_buffer;
+
+/**
+ * @brief Read-write property<string> to enable/disable CPU as acceleration (or helper device) at the beginning
```
Comment (riverlijunjie) [inline src/bindings/c/include/openvino/c/ov_property.h:184]:
```suggestion
 * @brief Read-write property<string> to enable/disable CPU as accelerator (or helper device) at the beginning
```

File: src/bindings/c/tests/ov_core_test.cpp [hunk 4]
```diff
@@ -572,4 +572,34 @@ TEST_P(ov_core_test, ov_core_compile_model_from_file_unicode) {
...
+TEST_P(ov_core_test_auto, ov_core_auto_set_and_get_property_bool) {
+    auto device_name = GetParam();
+    ov_core_t* core = nullptr;
+    OV_EXPECT_OK(ov_core_create(&core));
+    EXPECT_NE(nullptr, core);
+
+    const char* key_1 = ov_property_key_intel_auto_device_bind_buffer;
+    const char* key_2 = ov_property_key_intel_auto_enable_startup_fallback;
+    const char* key_3 = ov_property_key_intel_auto_enable_runtime_fallback;
+    const char* disable = "NO";
```
Comment (riverlijunjie) [inline src/bindings/c/tests/ov_core_test.cpp:587]: Why not add test code for value = "YES"?

Comment (WeldonWangwang) [reply src/bindings/c/tests/ov_core_test.cpp:587]: Because the default value is "YES", so set "NO" to test whether the setting is successful.

**Review by jiwaszki — APPROVED** [review]
Overall looks good on Python API side, just one style comment.

File: src/bindings/python/src/pyopenvino/core/properties/properties.cpp [hunk 5]
```diff
@@ -95,6 +95,15 @@ void regmodule_properties(py::module m) {
...
     wrap_property_RW(m_intel_gpu, ov::intel_gpu::enable_loop_unrolling, "enable_loop_unrolling");
 
+    // Submodule auto
+    py::module m_intel_auto =
+        m_properties.def_submodule("intel_auto",
+                                   "openvino.runtime.properties.intel_auto submodule that simulates ov::intel_auto");
+
+    wrap_property_RW(m_intel_auto, ov::intel_auto::device_bind_buffer, "device_bind_buffer");
+    wrap_property_RW(m_intel_auto, ov::intel_auto::enable_startup_fallback, "enable_startup_fallback");
+    wrap_property_RW(m_intel_auto, ov::intel_auto::enable_runtime_fallback, "enable_runtime_fallback");
```
[the remark this answers is not in the document]
Comment (WeldonWangwang) [reply src/bindings/python/src/pyopenvino/core/properties/properties.cpp]: > One nitpick: you can move it up or down a little as it gets in-between `m_intel_gpu` and another GPU-related `m_intel_gpu_hint`.

Modified.

File: src/bindings/c/include/openvino/c/ov_property.h [hunk 6]
```diff
@@ -172,6 +172,28 @@ ov_property_key_enable_profiling;
...
+ */
+OPENVINO_C_VAR(const char*)
+ov_property_key_intel_auto_enable_startup_fallback;
+
+/**
+ * @brief Read-write property<string> to enable/disable runtime fallback to other devices when infer fails
+ * @ingroup ov_property_c_api
+ */
+OPENVINO_C_VAR(const char*)
+ov_property_key_intel_auto_enable_runtime_fallback;
```
Comment (ilya-lavrenov) [inline src/bindings/c/include/openvino/c/ov_property.h:195]: @riverlijunjie do you think we need a dedicated file for non-common properties? Just not to keep everything is one file

Comment (riverlijunjie) [reply src/bindings/c/include/openvino/c/ov_property.h:195]: yes, I think so, which will make the header file more clear, similar with openvino/c/gpu/gpu_plugin_properties.h @WeldonWangwang do we have chance to create a dedicated file for auto plugin's properties of C header file before FF?

Comment (WeldonWangwang) [reply src/bindings/c/include/openvino/c/ov_property.h]: Hi @ilya-lavrenov, @riverlijunjie  Added dedicated files for auto plugin.

File: src/bindings/c/include/openvino/c/auto/auto_plugin_properties.h [hunk 7]
```diff
@@ -0,0 +1,34 @@
+// Copyright (C) 2018-2022 Intel Corporation
```
Comment (riverlijunjie) [inline src/bindings/c/include/openvino/c/auto/auto_plugin_properties.h:1]:
```suggestion
// Copyright (C) 2018-2023 Intel Corporation
```

File: src/bindings/c/tests/ov_auto_plugin_test.cpp [hunk 8]
```diff
@@ -0,0 +1,62 @@
...
+    EXPECT_NE(nullptr, core);
+
+    const char* key_1 = ov_property_key_intel_auto_device_bind_buffer;
+    const char* key_2 = ov_property_key_intel_auto_enable_startup_fallback;
+    const char* key_3 = ov_property_key_intel_auto_enable_runtime_fallback;
+    const char* enable = "YES";
+    const char* disable = "NO";
+    OV_EXPECT_OK(ov_core_set_property(core, device_name.c_str(), key_1, enable));
+    char* ret = nullptr;
+    OV_EXPECT_OK(ov_core_get_property(core, device_name.c_str(), key_1, &ret));
```
Comment (riverlijunjie) [inline src/bindings/c/tests/ov_auto_plugin_test.cpp:31]: If we go through all (key, value) pairs, why not use unified INSTANTIATE_TEST_SUITE_P to do it?

File: src/bindings/c/include/openvino/c/openvino.h [hunk 9]
```diff
@@ -30,3 +30,4 @@
 #include "openvino/c/ov_remote_context.h"
 #include "openvino/c/ov_shape.h"
 #include "openvino/c/ov_tensor.h"
+#include "openvino/c/auto/auto_plugin_properties.h"
```
Comment (ilya-lavrenov) [inline src/bindings/c/include/openvino/c/openvino.h:33]:
```suggestion
#include "openvino/c/auto/properties.h"
```
don't use `plugin` in names

File: src/bindings/c/tests/ov_auto_plugin_test.cpp (hunk 8 above)
Comment (WeldonWangwang) [reply src/bindings/c/tests/ov_auto_plugin_test.cpp]: Yes, updated.

**Review by riverlijunjie — APPROVED** [review]

File: src/bindings/c/tests/ov_auto_property_test.cpp [hunk 10]
```diff
@@ -0,0 +1,42 @@
...
+
+TEST_P(ov_auto_plugin_test, ov_core_auto_set_and_get_property_bool) {
+    ov_core_t* core = nullptr;
+    OV_EXPECT_OK(ov_core_create(&core));
+    EXPECT_NE(nullptr, core);
+    OV_EXPECT_OK(ov_core_set_property(core, device_name.c_str(), auto_property, property_value));
+    char* ret = nullptr;
+    OV_EXPECT_OK(ov_core_get_property(core, device_name.c_str(), auto_property, &ret));
+    EXPECT_STREQ(property_value, ret);
+    ov_free(ret);
```
Comment (wangleis) [inline src/bindings/c/tests/ov_auto_property_test.cpp:29]: Please add test case for invalid input value.

Comment (WeldonWangwang) [reply src/bindings/c/tests/ov_auto_property_test.cpp:35]: Added.

**Review by wangleis — APPROVED** [review]

**Pull request merged**